src/Computation/IContext.cs

```diff
@@ -5,6 +5,11 @@ namespace Symbolica.Computation;
 
 public interface IContext : IDisposable
 {
-    TResult Execute<TResult>(Func<Context, TResult> func)
-        where TResult : Z3Object;
+    Solver CreateSolver(Func<Context, Solver> func);
```

Comment:
The whole point of this PR is that I want to control creation of the solver within the context.